packages/firestore/src/util/misc.ts

```diff
@@ -28,8 +29,17 @@ export class AutoId {
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
-    for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+    while (autoId.length < 20) {
+      const bytes = PlatformSupport.getPlatform().randomBytes(40);
+      for (let i = 0; i < bytes.length; ++i) {
+        // Length of `chars` is 62. We only take bytes between 0 and 62*4-1
+        // (both inclusive). The value is then evenly mapped to indices of `char`
+        // via a modulo operation.
+        const maxValue = 62 * 4 - 1;
```

Comment:
Consider moving the calculation of maxValue to immediately follow the declaration of `chars`. IMO, having it re-calculated on each loop iteration detracts from readability because it incorrectly implies that its value may change between iterations.